src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1232,6 +1280,64 @@ because xbuild doesn't support framework reference assemblies.
 	</ResolveLibraryProjectImports>
 </Target>
 
+<Target Name="_CollectLibraryResourceDirectories">
+	<CollectNonEmptyDirectories Directories="@(LibraryResourceDirectories);@(_AdditonalAndroidResourceCachePaths->'%(Identity)\res')">
+		<Output TaskParameter="Output" ItemName="_LibraryResourceDirectories" />
+	</CollectNonEmptyDirectories>
+</Target>
+
+<Target Name="_CompileAndroidLibraryResources" DependsOnTargets="_CollectLibraryResourceDirectories"
+		Condition="'$(AndroidUseAapt2)' == 'True' And '$(DesignTimeBuild)' != 'True'"
+		Inputs="%(_LibraryResourceDirectories.FileFound)"
+		Outputs="%(_LibraryResourceDirectories.Identity)\..\compiled.stamp"
+	>
+	<ConvertResourcesCases
+		ResourceDirectories="@(_LibraryResourceDirectories)"
+		AcwMapFile="$(_AcwMapFile)"
+		AndroidConversionFlagFile="%(_LibraryResourceDirectories.Identity)\..\compiled.flata"
+		ContinueOnError="$(DesignTimeBuild)"
+	/>
+	<Aapt2Compile
+		ResourceDirectory="%(_LibraryResourceDirectories.Identity)"
+		OutputFlatArchive="%(_LibraryResourceDirectories.Identity)\..\compiled.flata"
+		ExplicitCrunch="$(AndroidExplicitCrunch)"
+		ToolPath="$(Aapt2ToolPath)"
+		ToolExe="$(Aapt2ToolExe)"
+		ContinueOnError="$(DesignTimeBuild)"
+	/>
+	<Touch Files="@(_LibraryResourceDirectories->'%(Identity)\..\compiled.stamp')" AlwaysCreate="True" />
+	<ItemGroup>
+		<FileWrites Include="%(_LibraryResourceDirectories.Identity)\..\compiled.flata" />
```

Comment:
This is what (more or less) caused Issue #1678: if `@(_LibraryResourceDirectories)` is *empty* (for whatever reason), *this is still evaluated*, meaning this would be equivalent to:

```xml
<FileWrites Include="\..\compiled.flata" />
```

I fear what that may do.

This should either be `Condition=" '@(_LibraryResourceDirectories)' != ''`, or this should instead be:

```xml
<FileWrites Include="@(_LibraryResourceDirectories->'%(Identity)\..\compiled.flata')" />
```